### PR TITLE
enable user controlled cde job submission parameters

### PR DESCRIPTION
**Pull request by tovganesh**
Enable the following job submission parameters to be controlled via profiles.yml file using cde_session_parameters:
<pre>
  cde_session_parameters:
       dbt.cde.job_timeout: 500 (default 900)
       dbt.cde.poll_wait: 15 (default 30)
       dbt.cde.log_wait: 20 (default 40) 
</pre>
All time parameters are in seconds.

### CODE REVIEW

File: dbt/adapters/spark_cde/cdeapisession.py [hunk 1]
```diff
@@ -738,6 +740,29 @@ def connect(self, user_name, password, base_auth_url, base_api_url, session_para
...
+                  DEFAULT_LOG_WAIT = int(value)
+            else:
+               new_session_params[key] = value
+
+          # remove adapter specific params
+          self.session_params = new_session_params
+        except Exception as e:
+          logger.info(f"Unable to set dbt-spark-cde parameters: {e}")
+
+        logger.debug(f"[spark-cde config] timeout = {DEFAULT_CDE_JOB_TIMEOUT}, poll_wait = {DEFAULT_POLL_WAIT}, log_wait = {DEFAULT_LOG_WAIT}")
```
Comment (sujitkp-blr) [inline dbt/adapters/spark_cde/cdeapisession.py:764]: Please just mention in debug that it is in seconds.

**Review by sujitkp-blr — DISMISSED** [review]
Looks fine

File: dbt/adapters/spark_cde/cdeapisession.py (hunk 1 above)
Comment (tovganesh) [reply dbt/adapters/spark_cde/cdeapisession.py]: @sujitkp-blr added this.

File: dbt/adapters/spark_cde/cdeapisession.py [hunk 2]
```diff
@@ -699,6 +699,8 @@ def get_auth_endpoint(self):
         return self.get_base_auth_url() + "gateway/authtkn/knoxtoken/api/v1/token"
 
     def connect(self, user_name, password, base_auth_url, base_api_url, session_params, verify_ssl_certificate=False):
+        global DEFAULT_CDE_JOB_TIMEOUT, DEFAULT_POLL_WAIT, DEFAULT_LOG_WAIT
```
Comment (TapasSenapati) [inline dbt/adapters/spark_cde/cdeapisession.py:702]: Looks good. Can we add some default values here in case user forgets to add.

**Review by TapasSenapati — CHANGES_REQUESTED** [review]

File: dbt/adapters/spark_cde/cdeapisession.py (hunk 2 above)
Comment (tovganesh) [reply dbt/adapters/spark_cde/cdeapisession.py:702]: defaults are already added in L41-48.

**Review by TapasSenapati — APPROVED** [review]

**Comment by tovganesh** [comment]
closing this replace by: https://github.com/cloudera/dbt-spark-cde/pull/58

**Pull request closed without merging**